### DISCUSSION TITLE
Trouble publishing to netlify after purchase of custom domain

### DISCUSSION THREAD

**Discussion by EllaKaye** [quarto-dev/quarto-cli#9992] [Q&A] [answered]
### Description

Hello. I have a Quarto website which I publish manually to Netlify. Today, I purchased a new domain name for the site and this seems to have messed up publishing. When I ran `quarto publish netlify` as usual, I was asked: `Publish update to: new-site-url` which I said yes to, and also confirmed the email account. Quarto then built the site fine, but then when it was getting ready to publish it, I got the following error message:

```
ERROR: ApiError: Generic Error

Stack trace:
    at catchErrorCodes (file:///Applications/quarto/bin/quarto.js:100453:15)
    at file:///Applications/quarto/bin/quarto.js:100474:17
    at eventLoopTick (ext:core/01_core.js:183:11)
```

I realised I hadn't updated `_publish.yml` with the new URL, so I did that and tried again but got the same error (though the site ID hasn't changed and the old netlify.app domain is still associated with it).

On the site's netlify page, it says the production is "prepared" and that "Netlify's robots are busy deploying your site to our CDN", but the Deploy log is blank except a spinning cursor (it's been spinning for 15 minutes and the site usually deploys in 2 seconds). I'm not sure if this is a Quarto issue or a Netlify one. 

Any idea what might be going on here and how to fix it?

**Comment by cderv** [comment]
Can you share your quarto version ? 

Also, could you try more debug log ? 
https://quarto.org/docs/troubleshooting/#verbose-mode

I don't know if we can have more info, but this is to know what does not work. We use Netlify npm client for the publishing. 
Hopefully, we are not hitting a limit with publishing to website with custom domain 🤔 

I don't really how we could reproduce to investigate this, so hopefully we'll get more information

**Reply by cscheid** [reply]
This is still a horrible error message. It comes from here:

https://github.com/quarto-dev/quarto-cli/blob/c3b39f746cd21b38f843cecd879af707f2efa7d4/src/publish/netlify/api/core/request.ts#L282-L284

We should do better than that.

**Reply by cderv** [reply]
Issue is now tracked at 
- https://github.com/quarto-dev/quarto-cli/issues/10036

**Comment by EllaKaye** [comment, marked as answer]
Thanks for your prompt response and suggestions for looking into this. The problem appears to have resolved itself, perhaps because I only registered the domain this morning, and it took a few hours for whatever behind-the-scenes changes were necessary for everything to work properly again.

**Reply by mcanouil** [reply]
Indeed, DNS cache can take several hours to refresh with new information.
Usually, host providers display a warning about this.
